modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaFieldWidgetTest.php

```diff
@@ -0,0 +1,236 @@
+    $display = \Drupal::service('entity_display.repository')
+      ->getViewDisplay($this->field->getTargetEntityTypeId(), $this->field->getTargetBundle())
+      ->setComponent($this->fieldStorage->getName(), $this->viewDisplayOptions);
+    $display->save();
+  }
+
+  /**
+   * Tests the featured media widget.
+   */
+  public function testFeaturedMediaWidget(): void {
+    // Visit the node add page.
+    $this->drupalGet('node/add/page');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    // Assert that the fields and labels are present on the page.
+    $assert_session->pageTextContains('Featured media');
+    $assert_session->fieldExists('Media item');
+    $assert_session->fieldExists('Caption');
+
+    // Assert the help texts without media overview permission.
+    $assert_session->pageTextContains('Type part of the media name.');
+    $assert_session->pageTextNotContains('See the media list (opens a new window) to help locate media.');
+    $assert_session->pageTextContains('Allowed media types: Image');
+
+    // Login with a user with extended permissions.
+    $this->drupalLogin($this->drupalCreateUser([
+      'access content',
+      'access media overview',
+      'create image media',
+      'create page content',
+    ]));
+    $this->drupalGet('node/add/page');
+
+    // Assert the help texts with media overview permission.
+    $assert_session->pageTextContains('Type part of the media name.');
+    $assert_session->pageTextContains('See the media list (opens a new window) to help locate media.');
+    $assert_session->pageTextContains('Allowed media types: Image');
+
+    // Test that the Media item field turns required once the Caption is filled.
+    $this->assertFalse($page->findField('Media item')->hasAttribute('required'));
+    $page->fillField('Caption', 'Caption text');
+    $this->assertTrue($page->findField('Media item')->hasAttribute('required'));
+
+    // Test the autocomplete functionality returns the created media items.
+    $this->doAutocomplete($this->fieldStorage->getName());
+    $results = $page->findAll('css', '.ui-autocomplete li');
+    $this->assertCount(2, $results);
+    $assert_session->pageTextContains('Test image');
+    $assert_session->pageTextContains('Test media');
+
+    // Assign the image media and save the node.
+    $page->fillField('Media item', 'Test image');
+    $page->fillField('Title', 'My test node');
+    $page->pressButton('Save');
+
+    // Assert the label and values are visible on the node page.
+    $assert_session->pageTextContains('Featured media field');
+    $assert_session->pageTextContains('Test image');
+    $assert_session->pageTextContains('Caption text');
+  }
+
+  /**
+   * Executes an autocomplete on a given field and waits for it to finish.
+   *
+   * @param string $field_name
+   *   The field name.
+   */
+  protected function doAutocomplete(string $field_name): void {
```

Comment:
What a missed opportunity to pass the actual value to search for in the autocomplete..